syntax = "proto3";
package proto;

message SymStackConst
{
    bool negate = 1;
    int32 offset = 2;
    int32 displacement = 3;
    bytes symbol_uuid = 4;
}

message SymAddrConst
{
    int64 displacement = 1;
    bytes symbol_uuid = 2;
}

message SymAddrAddr
{
    int64 scale = 1;
    int64 offset = 2;
    bytes symbol1_uuid = 3;
    bytes symbol2_uuid = 4;
}

message SymbolicExpression
{
    oneof value {
        SymStackConst stack_const = 1;
        SymAddrConst addr_const = 2;
        SymAddrAddr addr_addr = 3;
    }
}
